relay-filter/src/health_check.rs

```diff
@@ -0,0 +1,162 @@
+//! Implements event filtering based on whether the endpoint called is a healthcheck endpoint
+//!
+//! If this filter is enabled messages to healthcheck endpoints will be filtered out
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+use relay_general::protocol::Event;
+
+use crate::{FilterConfig, FilterStatKey};
+
+static HEALTH_CHECK_ENDPOINTS: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(
+        r#"(?ix)
+        healthcheck|
+        healthy|
+        live|
+        ready|
+        heartbeat|
+        /health$|
+        /healthz$
+        "#,
+    )
+    .expect("Invalid healthcheck filter Regex")
+});
```

Comment:
How often will this list get updated? Are we planning on new inbound filters on transaction names?

The health check filter checks transaction names, there's nothing specific to health checks besides this pattern. However, if we need to change the pattern, it requires redeploying Relay. Could we benefit from this filter if we make it more generic, and accept a pattern from Sentry? This is the approach with the [client IPs inbound filter](https://github.com/getsentry/relay/blob/dd93441bbb607dba806c6f1ca0c03783e348f048/relay-filter/src/client_ips.rs#L43). This will give us more flexibility. What do you think?

Reply:
I don't really know, we had a discussion and it was decided that we'd rather update Relay than add this to the configuration of each project ( I guess the implication is that it will not be updated very often).